src/store.cc

```diff
@@ -1726,6 +1735,8 @@ StoreEntry::startWriting()
     assert (isEmpty());
     assert(mem_obj);
 
+    clearPublicKeyScope();
```

Comment:
Please check whether we can improve "no `ksRevalidation` keys in cache" guarantees further by moving this call to `makePublic(ksDefault)`. Currently, calling `makePublic(ksDefault)` has no effect on `ksRevalidation` entries. I wonder whether we can improve (and arguably even simplify) code by changing that design?

Current PR description contains a  TODO-marked paragraph that assumes this change was implemented without surprises (and explains why this change is a good idea). We will need to adjust that paragraph as needed (or completely replace it if you find reasons to reject this change request).